### ISSUE TITLE
Propp-Wilson algorithm

### ISSUE THREAD

**Issue by MichaelClerx** [pints-team/pints#1117]
Seems fairly simple.

@ben18785 ?

**Comment by ben18785** [comment]
Yeah, it's an interesting one because, in theory, it allows perfect
sampling from the posterior. Got a feeling there are some practical issues
with it, however (but could be wrong), given that it was first proposed
about 34 years ago...

**Comment by MichaelClerx** [comment]
Agreed. I myself was first proposed some 37 years ago and have been encountering practical issues ever since.

I've discovered* this little book in my bookcase (https://www.cambridge.org/core/books/finite-markov-chains-and-algorithmic-applications/EE10AF27811B43B02E05905DC6413467), that has 3 short chapters on implementing this so might give it a go. (*bought it by accident years ago hoping it was about a different kind of markov chains)

**Comment by ben18785** [comment]
Haha. I almost wrote something like that about myself -- science doesn't
encourage optimistic thinking about one's usefulness!

Give it a go -- I'm intrigued to see how it does in any case!
